### PR TITLE
fix: ignore gamepads that are masked by steam input to prevent dublicated inputs

### PR DESCRIPTION
**Pull request by bmsuseluda**
<--- Put the description here --->
Fixes https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/issues/989
---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [X] Tested the feature and it's working on a current and clean install.
- [X] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

**Comment by flavioislima** [comment]
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/main/doc/cla.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   I have read the CLA Document and I hereby sign the CLA
   - - -
   <sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

**Comment by bmsuseluda** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Etaash-mathamsetty** [comment]
thanks for this, ill test soon

File: src/frontend/helpers/gamepad.ts [hunk 1]
```diff
@@ -300,13 +300,37 @@ export const initGamepad = () => {
...
+  }
+
+  function isMaskedGamepad(
+    valveGamepads: (Gamepad | null)[],
+    gamepad: Gamepad
+  ) {
+    return valveGamepads.find(
+      (valveGamepad) =>
+        valveGamepad &&
+        Math.abs(valveGamepad.timestamp - gamepad.timestamp) <= 10
```
Comment (Etaash-mathamsetty) [inline src/frontend/helpers/gamepad.ts:318]: why is this needed btw?

Comment (bmsuseluda) [reply src/frontend/helpers/gamepad.ts:318]: sadly the timestamps of the masked gamepad and steam input gamepad are not identical. most of the time there is a difference of 5, but sometimes it is up to 8. Therefore i used 10 to be sure to catch the masked gamepad. sadly the nearly identical timestamp is the only thing they have in common.

**Comment by arielj** [comment]
@bmsuseluda I'm testing this and it seems to work as expected, but there are some code check and test failures, can you check that?

**Comment by bmsuseluda** [comment]
@arielj thanks for the tip. the checks are green again 😃

**Review by arielj — APPROVED** [review]
thanks! I'll merge this
